src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/Configuration/JsonDataConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.DataSources.Json.Configuration;
+
+/// <summary>
+/// A configuration class for holding Json Data config values.
+/// </summary>
+public class JsonDataConfiguration
+{
+    private const string BaseDir = "data-files";
+    private const string DefaultDataFile = "test-data-azure-emissions.json";
+    private const string RegExDir = @"^[-/a-zA-Z_\d ]*$";
+    private string? dataFileLocation;
+    private string assemblyDirectory;
+
+    public const string Key = "JsonData";
+
+
+    public JsonDataConfiguration()
+    {
+        var assemblyPath = Assembly.GetExecutingAssembly().Location;
+        assemblyDirectory = Path.GetDirectoryName(assemblyPath)!;
+        DataFileLocation = DefaultDataFile;
+    }
+
+    /// <summary>
+    /// Json data file location
+    /// </summary>
+    public string DataFileLocation
+    {
+        get => dataFileLocation!;
+        set
+        {
```

Comment:
Just curious- why are we restrictive about the file location/name? Will this regex not allow numbers? Regex is always confusing to me ;)

Reply:
never mind, I got it after looking at the unit tests. :)